Source/Diffusion.cpp

```diff
@@ -247,20 +247,16 @@ Diffusion::diffuse_scalar (const Vector<MultiFab*>&  S_old,
       amrex::Print() << "... Diffusion::diffuse_scalar(): \n"
                      << " lev: " << level << '\n';
 
-    // Velocity components should go to tensor solver
-    if (S_comp <= Xvel && Xvel <= S_comp+nComp-1){
-       amrex::Abort("Diffusion::diffuse_scalar(): velocity component(s) attemping to use scalar solver. Velocity must use tensor solver.\n");
-    }
-
     // Check if scalars are diffusive type:
     // all the nComp scalars must be of the same type
     int diffType = a_is_diffusive[0];
-    for (int comp = 1; comp < nComp; comp++) {
-        if (a_is_diffusive[comp] != diffType) {
-            amrex::Abort("All the scalars must be either diffusive or non-diffusive when calling diffuse_scalar");
+    if(navier_stokes->do_tensor_visc){
```

Comment:
This if is not right. It's always required that all the scalars included here are either all diffusive or non-diffusive.

Reply:
It would be good to add a check at the beginning of `diffuse_scalar` that ensures that velocity is done on it's own and not with scalars. The way beta is set requires this.